src/pyfmi/fmi.pyx

```diff
@@ -2704,6 +2797,10 @@ cdef class FMUModelCS1(FMUModelBase):
         """
         Deallocate memory allocated
         """
+        self._invoked_dealloc = 1
+        if self._log_stream:
```

Comment:
Hmm, why is this done here? It seems like if you have a stream, you remove it directly?

Reply:
I thought this is what we agreed to during a separate discussion but I probably misunderstood this because I agree with the second comment also about the terminate/free methods. I've updated it to do this last instead (i.e. after we free self._fmu_log_name).